### ISSUE TITLE
failed to get container metrics when pod is Terminating 

### ISSUE THREAD

**Issue by guhan121** [kubernetes/kubernetes#118864]
### What happened?

i test hpa with `ContainerResource`,  when pod is Terminating i will got a error 
```
unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
```
if the pod alway is Terminating, it will cause the HPA not to work.

this is my test code 
```
package tests

import (
	"context"
	"github.com/stretchr/testify/require"
	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
	"k8s.io/client-go/discovery"
	cacheddiscovery "k8s.io/client-go/discovery/cached"
	"k8s.io/client-go/informers"
	"k8s.io/client-go/restmapper"
	"k8s.io/client-go/tools/cache"
	"k8s.io/kubernetes/pkg/controller"
	"k8s.io/kubernetes/pkg/controller/podautoscaler"
	metricsclient "k8s.io/kubernetes/pkg/controller/podautoscaler/metrics"
	"k8s.io/metrics/pkg/client/clientset/versioned"
	versionedCMC "k8s.io/metrics/pkg/client/custom_metrics"
	versionedEMC "k8s.io/metrics/pkg/client/external_metrics"
	"testing"
	"time"
)

const (
	// defaultTestingTolerance is default value for calculating when to
	// scale up/scale down.
	defaultTestingTolerance                     = 0.1
	defaultTestingCPUInitializationPeriod       = 2 * time.Minute
	defaultTestingDelayOfInitialReadinessStatus = 10 * time.Second
)

func TestMetricsGo(t *testing.T) {
	... // get clu with master and  KubeConfig
	cli, conf, _ := client.BuildClient(clu.Master, clu.KubeConfig)
	podMetricsClient, _ := versioned.NewForConfig(conf)
	discoveryCli, _ := discovery.NewDiscoveryClientForConfig(conf)
	//mapper,_ := dynamicmapper.NewRESTMapper(discoveryCli,time.Second*10)
	apiVersionsGetter := versionedCMC.NewAvailableAPIsGetter(discoveryCli)
	cachedClient := cacheddiscovery.NewMemCacheClient(discoveryCli)
	restMapper := restmapper.NewDeferredDiscoveryRESTMapper(cachedClient)
	testCMClient := versionedCMC.NewForConfig(conf, restMapper, apiVersionsGetter)
	testEMClient, _ := versionedEMC.NewForConfig(conf)
	metricsClient := metricsclient.NewRESTMetricsClient(podMetricsClient.MetricsV1beta1(), testCMClient, testEMClient)

	informerFactory := informers.NewSharedInformerFactory(cli, controller.NoResyncPeriodFunc())
	informer := informerFactory.Core().V1().Pods()

	replicaCalc := podautoscaler.NewReplicaCalculator(metricsClient, informer.Lister(), defaultTestingTolerance, defaultTestingCPUInitializationPeriod, defaultTestingDelayOfInitialReadinessStatus)

	stop := make(chan struct{})
	defer close(stop)
	informerFactory.Start(stop)
	if !cache.WaitForNamedCacheSync("HPA", stop, informer.Informer().HasSynced) {
		return
	}

	selector, err := metav1.LabelSelectorAsSelector(&metav1.LabelSelector{
		MatchLabels: map[string]string{"app": "qt-java-test-aws30-gray"},
	})
	if err != nil {
		require.Nil(t, err, "something went horribly wrong...")
	}

	for {
		got, _ := cli.CoreV1().Pods("devops").List(context.TODO(), metav1.ListOptions{LabelSelector: selector.String()})
		outReplicas, outUtilization, outRawValue, outTimestamp, err := replicaCalc.GetResourceReplicas(context.TODO(), int32(len(got.Items)), 80,
			"cpu", "devops", selector, "qt-java-test")
		t.Log("%%%%", int32(len(got.Items)), outReplicas, outUtilization, outRawValue, outTimestamp, err)

		outReplicas, outUtilization, outRawValue, outTimestamp, err = replicaCalc.GetResourceReplicas(context.TODO(), int32(len(got.Items)), 80,
			"cpu", "devops", selector, "")
		t.Log("====", int32(len(got.Items)), outReplicas, outUtilization, outRawValue, outTimestamp, err)
		time.Sleep(1 * time.Second)
	}
}



```

Below is my log，  when i kill `qt-java-test-aws30-gray-68cb97fbfd-zpg8q`  HPA return an error.
```
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 3 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 3 2 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 3 2 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 3 2 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 3 2 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 3 2 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 2 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:74: ==== 2 0 0 1 2023-06-26 17:36:25 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 2 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 0 0001-01-01 00:00:00 +0000 UTC unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
    metrics_test.go:74: ==== 3 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:70: %%%% 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
    metrics_test.go:74: ==== 2 1 11 22 2023-06-26 17:37:30 +0800 CST <nil>
```

### What did you expect to happen?

Can fix pods that are in a terminated state？

### How can we reproduce it (as minimally and precisely as possible)?

copy me test code, and kill a pod.

### Kubernetes version

<details>

```console
$ kubectl version
# paste output here

Client Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.10", GitCommit:"8152330a2b6ca3621196e62966ef761b8f5a61bb", GitTreeState:"clean", BuildDate:"2021-08-11T18:06:15Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"20", GitVersion:"v1.20.10", GitCommit:"8152330a2b6ca3621196e62966ef761b8f5a61bb", GitTreeState:"clean", BuildDate:"2021-08-11T18:00:37Z", GoVersion:"go1.15.15", Compiler:"gc", Platform:"linux/amd64"}
```

</details>

### Cloud provider

<details>
no, this a self built cluster.
</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
# paste output here
NAME="Ubuntu"
VERSION="16.04 LTS (Xenial Xerus)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 16.04 LTS"
VERSION_ID="16.04"
HOME_URL="http://www.ubuntu.com/"
SUPPORT_URL="http://help.ubuntu.com/"
BUG_REPORT_URL="http://bugs.launchpad.net/ubuntu/"
UBUNTU_CODENAME=xenial

$ uname -a
# paste output here
Linux k8s-master-670ba4fbb4 4.9.0-141-custom #19 SMP Wed Nov 28 20:32:27 CST 2018 x86_64 x86_64 x86_64 GNU/Linux

# On Windows:
C:\> wmic os get Caption, Version, BuildNumber, OSArchitecture
# paste output here
```

</details>

### Install tools

<details>

</details>

### Container runtime (CRI) and version (if applicable)

<details>

</details>

### Related plugins (CNI, CSI, ...) and versions (if applicable)

<details>

</details>

**Comment by tamilselvan1102** [comment]
The problem might be that you don't have [Kubernetes Metrics Server](https://github.com/kubernetes-sigs/metrics-server) installed.

**Comment by tamilselvan1102** [comment]
AWS has [this doc](https://docs.aws.amazon.com/eks/latest/userguide/metrics-server.html) that states that by default Metrics Server isn't installed on EKS clusters.
The following steps the doc advised for installing the server:

- Deploy the Metrics Server with the following command:

    kubectl apply -f https://github.com/kubernetes-sigs/metrics-server/releases/latest/download/components.yaml

- Verify that the metrics-server deployment is running the desired number of pods with the following command.

    kubectl get deployment metrics-server -n kube-system

**Comment by guhan121** [comment]
@tamilselvan1102 
```
root@k8s-master-670ba4fbb4:/home/qiantao# kubectl get deployment metrics-server -n kube-system
NAME             READY   UP-TO-DATE   AVAILABLE   AGE
metrics-server   1/1     1            1           609d

root@k8s-master-670ba4fbb4:/home/qiantao# kubectl get deployment metrics-server -n kube-system  metrics-server -oyaml |grep image:
                  f:image: {}
          image: kubesphere/metrics-server:v0.4.2
                  f:image: {}
          image: kubesphere/metrics-server:v0.4.2

```
now i update it to 0.6.3
```
root@k8s-master-670ba4fbb4:/home/qiantao# kubectl get deployment metrics-server -n kube-system  metrics-server -oyaml |grep image:
                  f:image: {}
          image: registry.cn-hangzhou.aliyuncs.com/yy_default/metrics-server:v0.6.3
                  f:image: {}
          image: registry.cn-hangzhou.aliyuncs.com/yy_default/metrics-server:v0.6.3
```

when i kill a pod 
```
qt-java-test-aws30-gray-68cb97fbfd-44lkj   0/1     ContainerCreating   0          1s
qt-java-test-aws30-gray-68cb97fbfd-c8dhf   1/1     Running             0          14m
qt-java-test-aws30-gray-68cb97fbfd-vplsp   0/1     Terminating         0          9s

```
now i don't receive a error like 
```
unable to get metrics for resource cpu: failed to get container metrics: container qt-java-test not present in metrics for pod devops/qt-java-test-aws30-gray-68cb97fbfd-zpg8q
```
the log like
```
  metrics_test.go:70: %%%% 3 0 0 1 2023-06-27 10:08:47 +0800 CST <nil>
    metrics_test.go:74: ==== 3 0 0 1 2023-06-27 10:08:47 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 0 0 1 2023-06-27 10:08:47 +0800 CST <nil>
    metrics_test.go:74: ==== 3 0 0 1 2023-06-27 10:08:47 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-27 10:08:47 +0800 CST <nil>
    metrics_test.go:74: ==== 3 2 0 1 2023-06-27 10:08:47 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-27 10:09:02 +0800 CST <nil>
    metrics_test.go:74: ==== 3 2 0 1 2023-06-27 10:09:02 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-27 10:09:02 +0800 CST <nil>
    metrics_test.go:74: ==== 3 2 0 1 2023-06-27 10:09:02 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-27 10:09:02 +0800 CST <nil>
    metrics_test.go:74: ==== 3 2 0 1 2023-06-27 10:09:02 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-27 10:09:02 +0800 CST <nil>
    metrics_test.go:74: ==== 3 2 0 1 2023-06-27 10:09:02 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-27 10:09:02 +0800 CST <nil>
    metrics_test.go:74: ==== 3 2 0 1 2023-06-27 10:09:02 +0800 CST <nil>
    metrics_test.go:70: %%%% 3 2 0 1 2023-06-27 10:09:02 +0800 CST <nil>
```

It looks like everything is working fine after the upgrade

thanks！

may be it fix on v0.6.0
